### PR TITLE
Fix nearby fonts for DxEngine again

### LINKED ISSUE #16322: warning popup "Cascadia Mono font is not found" (unpackaged version)
### Windows Terminal version

1.18.3181.0 (unpackaged 1.18.231114001)

### Windows build number

10.0.19045.0

### Other Software

_No response_

### Steps to reproduce

It looks like the bug fix for "The regression where "Cascadia Mono" (or "... Code") would not be found on upgrade has been un-regressed (https://github.com/microsoft/terminal/pull/16196)" introduced a regression for the unpackaged version

Launch WindowsTerminal.exe from unpackaged version
- https://github.com/microsoft/terminal/releases/download/v1.18.3181.0/Microsoft.WindowsTerminal_1.18.3181.0_x64.zip
- https://github.com/microsoft/terminal/releases/download/v1.18.2822.0/Microsoft.WindowsTerminal_1.18.2822.0_x64.zip

### Expected Behavior

- no warning popup from WindowsTerminal.exe unpackaged version 1.18.3181.0 (unpackaged 1.18.231114001)
- no warning popup from WindowsTerminal.exe unpackaged version 1.18.2822.0 (unpackaged 1.18.231009002-preview)

### Actual Behavior

- a warning popup: "Cascadia Mono font is not found, Consolas is used instead" from WindowsTerminal.exe unpackaged version 1.18.3181.0 (unpackaged 1.18.231114001) is displayed 
- no warning popup from WindowsTerminal.exe unpackaged version 1.18.2822.0 (unpackaged 1.18.231009002-preview)

with the same default profile from C:\...\AppData\Local\Microsoft\Windows Terminal\settings.json settings
```json
{
    "altGrAliasing": true,
    "antialiasingMode": "grayscale",
    "closeOnExit": "automatic",
    "colorScheme": "Campbell",
    "commandline": "C:\\...\\PowerShell-7.3\\pwsh.exe",
    "cursorShape": "bar",
    "font":
    {
        "face": "Cascadia Mono",
        "size": 10.0
    },
    "guid": "....",
    "hidden": false,
    "historySize": 9001,
    "icon": "C:\\...\\PowerShell-7.3\\assets\\Square44x44Logo.png",
    "name": "Pwsh 7.3",
    "padding": "8, 8, 8, 8",
    "snapOnInput": true,
    "startingDirectory": "%USERPROFILE%",
    "useAcrylic": false
},

### PR DESCRIPTION
**Pull request by lhecker**
The nearby font loading has to be outside of the try/catch of the
`_FindFontFace` call, because it'll throw for broken font files.
But in my previous PR I had overlooked that the font variant loop
modifies the only copy of the face name that we got and was in the
same try/catch. That's bad, because once we get to the nearby search
code, the face name will be invalid. This commit fixes the issue by
wrapping each individual `_FindFontFace` call in a try/catch block.

Closes #16322

## Validation Steps Performed
* Remove every single copy of Windows Terminal from your system
* Manually clean up Cascadia .ttf files because they aren't gone
* Destroy your registry by manually removing appx references (fun!)
* Put the 4 Cascadia .ttf files into the Dev app AppX directory
* Launch
* No warning ✅

### CODE REVIEW

File: src/renderer/dx/DxFontInfo.cpp [hunk 1]
```diff
@@ -126,44 +126,52 @@ void DxFontInfo::SetFromEngine(const std::wstring_view familyName,
     try
     {
         face = _FindFontFace(localeName);
+    }
+    CATCH_LOG();
```
Comment (lhecker) [inline src/renderer/dx/DxFontInfo.cpp:130]: Suppressing whitespace changes makes this a lot easier.
tl;dr: This PR wraps every single `_FindFontFace` call into its own try/catch block and moves the font-variant search code down below the nearby search.

**Review by DHowett — APPROVED** [review]

**Review by zadjii-msft — APPROVED** [review]
